### ISSUE TITLE
Default Route Dropping

### ISSUE THREAD

**Issue by spark1313** [opnsense/core#6667]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

After upgrading to version 23.1.6 from 22.7.11, I noticed odd issues occurring with internet connectivity and DNS. After a long time troubleshooting and trying suggestions in forum posts, I managed to get intermittent internet access and upgraded to 23.1.11 in hopes the bug would be fixed. Unfortunately, the issue persists on that version which I am currently on. I have switched around some settings and did process of elimination and found the specific circumstances.

After a reboot of the system, everything loads up fine and I am able to log in. Things remain that way for around 20s-60s and then the default route is dropped from the route table under System > Routes > Status. This kills the internet connection and it does not return.

Re-adding the route but as a static route seems to fix the issue but then upon reboot, it still drops the route and the issue persists until removing the static route, rebooting again, waiting for the default route to drop, then re-adding the static route.

I also found that if I reload the WAN interface in Interfaces > Overview, then the default route gets reassigned and stays until reboot. The same result is achieved by restarting the routing service under System > Diagnostic > Services.

**To Reproduce**

1. Reboot
2. Wait 20-60s

**Expected behavior**

The default route does not drop itself.

**Additional context**

I looked through the logs but don't think I saw anything significant. I may have been looking in the wrong section? If you know where I could pull logs for the default route dropping please let me know and I can check again.

Note that my setup is fairly vanilla. I'm new to OPNSense coming from pfSense land. I'm using Unbound for DNS. I don't use Adguard as some others experiencing similar issues did. I have one LAN static and one WAN DHCP interface set up with the router connected to an ONT box. I'm using intel NICs. I am using IPv4 only.

**Comment by fichtner** [comment]
I'm tagging this „incomplete“ since there is no log or configuration attached which makes your conclusion impossible to reproduce or trace.

**Comment by spark1313** [comment]
Hi @fichtner , apologies for the delay and lack of logs. I've ran through an isolated play of the drop occurring and produced a log at debug verbosity. Please see attached.
[log.txt](https://github.com/opnsense/core/files/12091909/log.txt)

**Comment by spark1313** [comment]
Also I did a bit more troubleshooting and confirmed that pulling the WAN cable and replacing readds the route and it doesnt drop again until reboot. Additionally, I tried with and without Default Gateway Switching, and Gateway Monitoring enabled and they had no effect on the route drop.

**Comment by MaxXor** [comment]
Sound similar to my issue, but my default route is missing immediately after reboots... https://github.com/opnsense/core/issues/6660 not 100% sure if it's the same issue, but definitely sounds related.

**Comment by spark1313** [comment]
> Sound similar to my issue, but my default route is missing immediately after reboots... #6660 not 100% sure if it's the same issue, but definitely sounds related.

Hmm, yeah looks like it may be similar. Possibly the same root cause but the surface level focus different. I tried with and without gateway monitoring but the results remained the same, although I didn't specify a gateway monitoring IP, just unchecked the box to enable gateway monitoring.

Thinking about it over another night and looking at your thread, it may indeed be some kind of race condition? It looks like upon reboot, the WAN interface first comes online and pulls an IP from existing routes and cached info, then something is triggered to reset WAN DHCP and pull new info which blasts away the default route and it is not re-added for some reason.

Also thought I'd mention that before experiencing this on 23.1.6, I upgraded from within the UI from a few versions back (a 20 or 21 release I believe) and kept the same config so that may have some relation? You mentioned you had 2 boxes, one working and one not, with it looks like the same config. Was one of them fresh and another an upgrade from a previous version and keeping the configuration by chance?

**Comment by spark1313** [comment]
Pulling the WAN cable seems to be inconsequential to re-establish the default route though, its just another method I came across. Restarting the routing service or refreshing a related configuration all seems to trigger some kind of script in the system which re-establishes the route.

**Comment by spark1313** [comment]
So I have an exciting update! I have simultaneously been troubleshooting why I was only getting 100baseT at my new location I am setting up when I subscribe to 1000baseT WAN.

I tried a bunch of things including bypassing the router connecting directly into the ONT box, checking cable run lengths and specs, trying different NICs, trying different ports, with and without switch, different patch cables, etc.

In the end, I did just did a brand new cable run which was needed anyway with a sufficiently short length but noticed it still not negotiating gigabit. When I retraced the wiring, I ended up at a ethernet replicator/extender device which used to extend the old run since the cable was a bit short which I forgot to remove. When I removed that from the chain and went straight off my new run, gigabit was successfully negotiated. Strange since I've used that same brand of device before and actually currently use one at a different location which negotiates gigabit no problem. Although that location uses much shorter runs of CAT7 and CAT6 while this one uses a lengthy CAT5e so maybe that had some kind of adverse effect on signal quality.

Anyway, can confirm that this appeared to be a hardware issue caused by an ethernet replicator/extender device which in turn caused a software issue. Strange that it works fine at 100baseT when the default route is in the table but somehow that crappy device caused the ONT to not only negotiate a lower speed but apparently also drop and reconnect.

@MaxXor Perhaps go sanity check your hardware chain leading to the router. Even if it doesnt negotiate down, maybe try a different switch, NIC, or cable? And make sure you aren't using an ethernet extender ;)
